stactools_landsat/stactools/landsat/utils.py

```diff
@@ -102,40 +103,33 @@ def transform_stac_to_stac(item: Item,
 
     if enable_proj:
         try:
-            # If we can load the blue band, use it to add proj information
-            if item.assets.get("SR_B2.TIF"):
-                asset = item.assets["SR_B2.TIF"]
-            # SR_B10 is a fallback for SR_B2
-            elif item.assets.get("SR_B10.TIF"):
-                asset = item.assets["SR_B10.TIF"]
-            else:
-                raise ValueError('Asset SR_B2.TIF or SR_B2.TIF required')
-
-            with rasterio.open(asset.href) as opened_asset:
-                shape = [opened_asset.height, opened_asset.width]
-                transform = opened_asset.transform
-                crs = opened_asset.crs.to_epsg()
-
-            # Now we have the info, we can make the fields
-            item.ext.enable("projection")
-            item.ext.projection.epsg = crs
-
+            retrieved_tile_info = None
             for name, asset in item.assets.items():
-                if asset.media_type == "image/vnd.stac.geotiff; cloud-optimized=true":
+                if "geotiff" in asset.media_type:
+                    if not retrieved_tile_info:
+                        with rasterio.open(asset.href) as opened_asset:
+                            shape = opened_asset.shape
+                            transform = opened_asset.transform
+                            crs = opened_asset.crs.to_epsg()
+                            retrieved_tile_info = True
+
                     item.ext.projection.set_transform(transform, asset=asset)
                     item.ext.projection.set_shape(shape, asset=asset)
                     asset.media_type = MediaType.COG
 
-        except RasterioIOError:
-            print("Failed to load blue band, so not handling proj fields")
+            # Now we have the info, we can make the fields
+            item.ext.enable("projection")
+            item.ext.projection.epsg = crs
```

Comment:
This should raise an exception if `crs` (and `shape` and `transform`) are unset.

Reply:
Pete,

It will raise an exception much before ```crs```. When the code tries to set ```shape``` and it's not defined it will raise the exception. 
What I can do and I'll do is add a better explanation(exception message) to this situation.

Reply:
Added new check, which raises an exception in case of missing any of those.